pandas/core/arraylike.py

```diff
@@ -157,11 +157,67 @@ def array_ufunc(self, ufunc: Callable, method: str, *inputs: Any, **kwargs: Any)
     --------
     numpy.org/doc/stable/reference/arrays.classes.html#numpy.class.__array_ufunc__
     """
+    from pandas.core.frame import DataFrame
     from pandas.core.generic import NDFrame
     from pandas.core.internals import BlockManager
 
     cls = type(self)
 
+    is_ndframe = [isinstance(x, NDFrame) for x in inputs]
```

Comment:
What is `is_series` ?

Reply:
we have dataframes and series

Reply:
Yes, and `NDFrame` is the parent class for both? Do you want me to put `isinstance(x, (Series, DataFrame))` instead of `isinstance(x, NDFrame)` ?

Reply:
yes i think its more clear

Reply:
Note that below in this `array_ufunc` function, we are also using `NDFrame` for this purpose

Reply:
so rename this to is_series_or_frame i think is more clear

Reply:
I renamed it now to `n_alignable`, because `alignable` is the variable name that is already used below, for consistency. And it also matches the explanation in the comment (which says this is Series or DataFrame). 
(but can also rename to `n_series_or_frame` if you prefer)